### PR TITLE
en: add source-configuration-file.md for 2.0

### PR DESCRIPTION
**Pull request by toutdesuite**
### What is changed, added or deleted? (Required)

add source-configuration-file.md for 2.0

### Which TiDB Data Migration (DM) version(s) do your changes apply to? (Required)

- [x] master (the latest development version, including v2.0 changes for now)
- [ ] v1.0 (TiDB DM 1.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-dm/pull/27
- Other reference link(s):

### CODE REVIEW

File: en/source-configuration-file.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+# Enable GTID
+
+enable-gtid = false
+
+[from]
+host = "127.0.0.1"
+port = 3306
+user = "root"
+password = "ZqMLjZ2j5khNelDEfDoUhkD5aV5fIJOe0fiog9w=" # The password here refers to the dmctl-encrypted password of the upstream MySQL user
```
Comment (TomShawn) [inline en/source-configuration-file.md:26]: What about `The password of the upstream user that is encrypted using dmctl`?

File: en/source-configuration-file.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+```
+
+## Configuration parameters
+
```
Comment (TomShawn) [inline en/source-configuration-file.md:30]:
```suggestion

This section describes each configuration parameter in the configuration file.

```

File: en/source-configuration-file.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+### Global configuration
+
+| Parameter | Description |
+| :------------ | :--------------------------------------- |
+| `source-id` | Identifies a MySQL instance. |
```
Comment (TomShawn) [inline en/source-configuration-file.md:35]:
```suggestion
| `source-id` | Marks a MySQL instance. |
```

File: en/source-configuration-file.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+| `enable-gtid` | Determines whether to pull binlog from the upstream using GTID. The default value is `false`. In general, you don't need to configure `enable-gtid` manually. However, if GTID is enabled in the upstream database, and the master/slave switch is required, you need to set `enable-gtid` to `true`. |
```
Comment (TomShawn) [inline en/source-configuration-file.md:36]:
```suggestion
| `enable-gtid` | Determines whether to pull binlogs from the upstream using GTID. The default value is `false`. In general, you do not need to configure `enable-gtid` manually. However, if GTID is enabled in the upstream database, and the master/slave switch is required, you need to set `enable-gtid` to `true`. |
```

File: en/source-configuration-file.md [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+| `host` | Specifies the host of the upstream database. |
+| `port` | Specifies the port of the upstream database. |
+| `user` | Specifies the username of the upstream database. |
+| `password` | Specifies the user password of the upstream database. Note that the password here refers to the dmctl-encrypted password. |
```
Comment (TomShawn) [inline en/source-configuration-file.md:40]:
```suggestion
| `password` | Specifies the user password of the upstream database. Note that the password must be encrypted using dmctl. |
```

**Comment by TomShawn** [comment]
@toutdesuite Please add this document in TOC.

**Comment by toutdesuite** [comment]
@TomShawn https://github.com/pingcap/docs-dm/pull/38 This PR gets both the Chinese doc and the English doc added into TOC

File: en/source-configuration-file.md (hunk 1 above)
Comment (toutdesuite) [reply en/source-configuration-file.md:26]: How about `The user password of the upstream database. Note that the password must be encrypted using dmctl.`?

**Comment by toutdesuite** [comment]
@TomShawn PTAL again. Thanks~

File: en/source-configuration-file.md (hunk 1 above)
Comment (TomShawn) [reply en/source-configuration-file.md]: Good.

File: en/source-configuration-file.md [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+## Configuration parameters
+
+This section describes each configuration parameter in the configuration file.
+
+### Global configuration
+
+| Parameter | Description |
+| :------------ | :--------------------------------------- |
+| `source-id` | Represents a MySQL instance ID. |
+| `enable-gtid` | Determines whether to pull binlogs from the upstream using GTID. The default value is `false`. In general, you do not need to configure `enable-gtid` manually. However, if GTID is enabled in the upstream database, and the master/slave switch is required, you need to set `enable-gtid` to `true`. |
```
Comment (TomShawn) [inline en/source-configuration-file.md:37]:
```suggestion
| `enable-gtid` | Determines whether to pull binlog from the upstream using GTID. The default value is `false`. In general, you do not need to configure `enable-gtid` manually. However, if GTID is enabled in the upstream database, and the master/slave switch is required, you need to set `enable-gtid` to `true`. |
```

**Pull request merged**